src/scripts/Baseball.gd

```diff
@@ -1,10 +1,60 @@
-extends Node2D
+class_name Baseball
+extends KinematicBody2D
 
+export var TIME_UNTIL_DISAPPEARS = 10.0
+export var BASEBALL_GRAVITY = 1.0 
+export var SMALL_BALL_FRAMES = 40
 
+onready var sprite = $Sprite
+onready var collision_shape = $CollisionShape2D
+
+var is_following_path = false
+var current_velocity: Vector2
+var current_frame = 0
 # Called when the node enters the scene tree for the first time.
 func _ready():
+	#print("Setting timeout")
+	var timer := Timer.new()
+	add_child(timer)
+	timer.one_shot = true
+	timer.wait_time= TIME_UNTIL_DISAPPEARS
+	timer.start()
+	timer.connect("timeout", self, "on_timeout")
+	
+	#Disable collisions for first X 'frames'
+	collision_shape.disabled = true 
+	sprite.scale = Vector2(0.5, 0.5)
+	
+	
 	pass # Replace with function body.
 
-# TODO
-func launch():
+
+func _physics_process(_delta):
+	#tracks invuln and scaling
+	current_frame +=1
```

Comment:
I feel like we can use a separate timer here instead of manually counting frames to keep track of invuln to make it easier to work with and tweak!

Reply:
hahaha this is both a holdover from me writing my own timers all the time, and also I tried to use the frame count to scale.

I was thinking of doing something like baseball.set_scale(currentframe/40, currentframe/40) so the ball would get progressively bigger, thinking that the set scale function had to be called every frame to maintain it relative to its initial scale. that is absolutely not how it works though, so the ball was almost immediately shrinking down to oblivion lol.

yeah this should definitely be a timer, I think that it would be cool to have the scale adjust progressively but theres probably a lot more graceful ways of handling that lol.

Reply:
hahaha ohhh gotcha gotcha, that makes sense!